modules/clipboard.js

```diff
@@ -68,4 +139,55 @@ class Clipboard extends Module {
 }
 
 
+function matchAliases(node, delta) {
+  let formats = {};
+  switch(node.tagName) {
+    case 'B':
+      formats = { bold: true };
+      break;
+    case 'I':
+      formats = { italic: true };
+      break;
+    default: return delta;
+  }
+  return delta.compose(new Delta().retain(delta.length(), formats));
+}
+
+function matchBlot(node, delta) {
+  let match = Parchment.query(node);
+  if (match == null) return delta;
+  if (match.prototype instanceof Parchment.Embed) {
+    let embed = {};
+    embed[match.blotName] = match.value(node);
+    delta.insert(embed, match.formats(node));
+  } else if (typeof match.formats === 'function') {
+    delta = delta.compose(new Delta().retain(delta.length(), match.formats(node)));
+  }
+  return delta;
+}
+
+function matchNewline(node, delta) {
+  if (!(node instanceof HTMLElement)) return delta;
+  if (BLOCK_ELEMENTS[node.tagName] || node.style.display === 'block') {
+    let lastOp = delta.ops[delta.ops.length - 1];
+    let endText = (lastOp == null || typeof lastOp.insert !== 'string') ? '' : lastOp.insert;
+    if (!endText.endsWith('\n')) {
+      delta.insert('\n');
+    }
+    if (node.style.paddingBottom || (node.style.marginBottom && !endText.endsWith('\n\n'))) {
```

Comment:
I was trying to avoid this as I recalled reading somewhere getComputedStyle was expensive and this function would be called on many of the content pasted, but testing it in JSFiddle just now, it does not seem prohibitively so..